src/Microsoft.SqlTools.ServiceLayer/FileBrowser/FileBrowserService.cs

```diff
@@ -132,7 +133,12 @@ public void InitializeService(ServiceHost serviceHost)
             FileBrowserCloseResponse response = new FileBrowserCloseResponse();
             FileBrowserOperation removedOperation;
             response.Succeeded = ownerToFileBrowserMap.TryRemove(fileBrowserParams.OwnerUri, out removedOperation);
-
+            if (removedOperation != null)
+            {
+                removedOperation.Cancel();
+                removedOperation.Dispose();
```

Comment:
so how do you know at this point if the operation canceled and it's safe to dispose?